src/Engine/GameEvents/UnloadEvent.java

```diff
@@ -50,8 +50,19 @@ public void performEvent(GameMap gameMap)
       System.out.println("WARNING! Failed to unload unit due to preconditions not being met:");
       if( transport.heldUnits == null ) System.out.println("          Transport unit is empty.");
       if( !transport.heldUnits.contains(cargo) ) System.out.println("          Unit to debark is not on transport.");
-      if( gameMap.getLocation(dropLoc).getResident() != null ) System.out.println("          Unload location is not empty.");
+      if( gameMap.getLocation(dropLoc).getResident(gameMap) != null ) System.out.println("          Unload location is not empty.");
     }
   }
-
+  
+  @Override
+  public boolean shouldPreempt(MapMaster gameMap )
+  {
+    boolean result = false;
+    Unit obstacle = gameMap.getLocation(dropLoc).getResident(gameMap);
+    if( null != obstacle && transport.CO.isEnemy(obstacle.CO) )
```

Comment:
If `obstacle` is friendly, then we don't preempt? Shouldn't it still be impossible to drop a unit onto an ally? Granted, that probably couldn't happen because you would see them to start with, but still.

Reply:
Hmm. If I try to unload three things, and the second one is blocked, the first one should still get out, right? We currently store drop locations as a `Map<Unit, XYCoord>`. We should consider making that an ordered list to preserve drop order.